schema/bool_test.go

```diff
@@ -176,6 +176,14 @@ func TestBoolSerializationCycle(t *testing.T) {
 			if serialized != output {
 				t.Fatalf("Invalid value after serialization: %v", serialized)
 			}
+
+			unserialized2, err := boolType.Unserialize(serialized)
+			assert.NoError(t, err)
+			serialized2, err := boolType.Serialize(unserialized2)
+			assert.NoError(t, err)
+			// test unserialize and serialize are reversible
+			assert.Equals(t, unserialized2, output)
+			assert.Equals(t, serialized2, serialized)
```

Comment:
Ditto my previous comment:  I don't think we need to reserialize the doubly unserialized value, because we've already done that, successfully.

Also, while I know that we're sensitive to feature creep, it seems like lines 158-178 should be changed to use the `assert.NoError()` and `assert.Equals()` functions, while you're here.  (Aside from improving the code quality, it would make the result (more) directly comparable to, e.g., the code in `any_test.go`.)

Reply:
removed